tasks/github_tasks.py

```diff
@@ -64,65 +65,48 @@ def trigger_macos_build(
-        version_cache_file_content=version_cache,
-    )
-
-
-@task
-def trigger_macos_test(
-    _,
-    datadog_agent_ref=DEFAULT_BRANCH,
-    release_version="nightly-a7",
-    python_runtimes="3",
-    destination=".",
-    version_cache=None,
-    retry_download=3,
-    retry_interval=10,
-):
-    _trigger_macos_workflow(
-        release_version,
-        destination,
-        retry_download,
-        retry_interval,
-        workflow_name="test.yaml",
-        datadog_agent_ref=datadog_agent_ref,
-        python_runtimes=python_runtimes,
-        version_cache_file_content=version_cache,
-    )
-
-
-@task
-def trigger_macos_lint(
-    _,
-    datadog_agent_ref=DEFAULT_BRANCH,
-    release_version="nightly-a7",
-    python_runtimes="3",
-    version_cache=None,
-):
-    _trigger_macos_workflow(
-        release_version,
-        workflow_name="lint.yaml",
-        datadog_agent_ref=datadog_agent_ref,
-        python_runtimes=python_runtimes,
-        version_cache_file_content=version_cache,
-    )
+    if workflow_type == "build":
+        conclusion = _trigger_macos_workflow(
+            # Provide the release version to be able to fetch the associated
+            # macos-build branch from release.json for all workflows...
+            release_version,
+            destination,
+            retry_download,
+            retry_interval,
+            workflow_name="macos.yaml",
+            datadog_agent_ref=datadog_agent_ref,
+            # ... And provide the release version as a workflow input when needed
+            release_version=release_version,
+            major_version=major_version,
+            python_runtimes=python_runtimes,
+            # Send pipeline id and bucket branch so that the package version
+            # can be constructed properly for nightlies.
+            gitlab_pipeline_id=os.environ.get("CI_PIPELINE_ID", None),
+            bucket_branch=os.environ.get("BUCKET_BRANCH", None),
+            version_cache_file_content=version_cache,
+        )
+    elif workflow_type == "test":
+        conclusion = _trigger_macos_workflow(
+            release_version,
+            destination,
+            retry_download,
+            retry_interval,
+            workflow_name="test.yaml",
+            datadog_agent_ref=datadog_agent_ref,
+            python_runtimes=python_runtimes,
+            version_cache_file_content=version_cache,
+        )
+        repack_macos_junit_tar(conclusion, "junit-tests_macos.tgz", "junit-tests_macos-repacked.tgz")
+    elif workflow_type == "lint":
+        conclusion = _trigger_macos_workflow(
+            release_version,
+            workflow_name="lint.yaml",
+            datadog_agent_ref=datadog_agent_ref,
+            python_runtimes=python_runtimes,
+            version_cache_file_content=version_cache,
+        )
+    if conclusion == "failure":
+        raise Exit(message=f"macos {workflow_type} workflow failed", code=1)
```

Comment:
If I understand correctly if the Github Action is canceled, the job on Gitlab will still report as successful. I am not sure this something we want. To me this is strange to have the job green while the build could fail but was canceled.

Another issue with that is that any job that depends on this one will be executed and will fail because the artifacts were not generated by the build. For example `deploy_packages_dmg-x64-a7`
Ideally we'd like the job to be displayed as `canceled` on gitlab as well, this is probably impossible. But IMHO if I had to chose I would keep the job as failed when the Github Action is canceled